src/mscorlib/shared/System/Text/StringBuilder.cs

```diff
@@ -436,6 +457,7 @@ public string ToString(int startIndex, int length)
 
         public StringBuilder Clear()
         {
+            Capacity = Math.Max(DefaultCapacity, Math.Min(Capacity, (int)(Length * 1.2)));
```

Comment:
needs a comment for your successor developers, eg something like
```c#
// Retain enough storage to store the same size content, plus 20% buffer, to help avoid reallocating during repeated use
```
We do'nt need to mention "and also because currently using Insert can lead to pathologically large allocations" ...